src/clr-angular/popover/common/abstract-popover.ts

```diff
@@ -114,11 +114,13 @@ export abstract class AbstractPopover implements AfterViewChecked, OnDestroy {
   private ignore: any;
 
   private attachESCListener(): void {
-    this.documentESCListener = this.renderer.listen('document', 'keydown', event => {
-      if (event && event.keyCode === ESC) {
-        this.ifOpenService.open = false;
-      }
-    });
+    if (!this.popoverOptions.ignoreGlobalESCListener) {
+      this.documentESCListener = this.renderer.listen('document', 'keydown', event => {
+        if (event && event.keyCode === ESC) {
```

Comment:
I believe `KeyboardEvent.keyCode` is deprecated: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

Recommended alternative is: `KeyboardEvent.key`

Reply:
Updated!

Reply:
Good to know, thanks Shijir. I also switched to key (from code) in my directive, though for different reasons.